pkg/manager/member/tidb_member_manager.go

```diff
@@ -301,6 +301,8 @@ func (tmm *tidbMemberManager) syncTiDBService(tc *v1alpha1.TidbCluster) error {
 	}
 	oldSvc := oldSvcTmp.DeepCopy()
 	util.RemainNodeport(newSvc, oldSvc)
+	// Remain healthCheckNodePort if it has been generated by apiserver
+	newSvc.Spec.HealthCheckNodePort = oldSvc.Spec.HealthCheckNodePort
```

Comment:
merge it with `util.RemainNodeport`? we can have a function to handle all details in service reconciliation and other places should use this function too to avoid poential issues in the future.

Reply:
agreed

Reply:
addressed in https://github.com/pingcap/tidb-operator/pull/2438/commits/4d6bda7096a2acec9fdd78bd0d6a551758740c60